src/gurt/dlog.c

```diff
@@ -483,14 +495,19 @@ d_log_sync(void)
 
 	if (mst.log_fd >= 0) {
 		rc = fsync(mst.log_fd);
-		if (rc < 0)
-			dlog_print_err(errno, "failed to sync log file\n");
+		if (rc < 0) {
+			int err = errno;
+
+			dlog_print_err(err, "failed to sync log file %d\n", mst.log_fd);
```

Comment:
Another idea here would be to not even try calling fsync if d_log_write() failed.  The stderr logging has rate-limiting however that won't work for when you get "pairs" of error logs, if the write failed then attempting the sync seems to be of little value.

Reply:
Done.